testapi.pm

```diff
@@ -2063,11 +2063,11 @@ in the corresponding variable
 
 sub data_url($) {
     my ($name) = @_;
-    if ($name =~ /^REPO_\d$/) {
+    if ($name =~ /^REPO_\d+$/) {
         return autoinst_url("/assets/repo/" . get_var($name));
     }
-    if ($name =~ /^ASSET_\d$/) {
-        return autoinst_url("/assets/other/" . get_var($name));
+    if ($name =~ /^ASSET_\d+$/) {
+        return autoinst_url("/assets/other/" . basename(get_var($name)));
```

Comment:
This won't work in all cases. It works on OSD for now because this code effectively bypasses the cache service and falls back to the shared asset directory mounted via NFS. But the point of the `ASSET_*` changes was to get one step closer to eliminating the NFS mount.

Basically, you'll need to check whether the absolute path points to `ASSETDIR`, or pool directory (CWD). You'll have to return different URLs depending on which path it is. And you'll also need to implement a new URL handler for assets in the pool directory in `commands.pm`.

Reply:
Thanks for the deeper insight. Let me check the code